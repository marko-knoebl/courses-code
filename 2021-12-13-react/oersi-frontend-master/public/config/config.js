window['runTimeConfig'] = {
  ELASTIC_SEARCH: {
    URL: "http://localhost:9200/",
    CREDENTIALS: "",
    APP_NAME: "oer_data"
  },
  GENERAL_CONFIGURATION:{
    PUBLIC_URL: "http://localhost/resources",
    RESULT_PAGE_SIZE_OPTIONS:["12", "24", "48", "96"],  // page size options configuration    
    NR_OF_RESULT_PER_PAGE:12,  //  number of results to show per view. Defaults to 12.
    /**
     * Accept a list of objects 
     * example:
     * {'path': 'public/{folderName}/{languageCode}/{fileName}.html', 'language': '{languageCode}'}
     * 
     */
    PRIVACY_POLICY_LINK: [],
    I18N_CACHE_EXPIRATION: 600000, // expiration time of the i18n translation cache storage
    I18N_DEBUG: false,
    TRACK_TOTAL_HITS: true, // track number of total hits from elasticsearch - see https://www.elastic.co/guide/en/elasticsearch/reference/7.10/search-your-data.html#track-total-hits
    FEATURES: {
      EMBED_OER: true, // feature toggle: use "embed-oer" button
      SCROLL_TOP_BUTTON: true, // feature toggle: use "scroll-to-top" button
      USE_RESOURCE_PAGE: true // feature toggle: use a single html-page per resource
    }
  }
}