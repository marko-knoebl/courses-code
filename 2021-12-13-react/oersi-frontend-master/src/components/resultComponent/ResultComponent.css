.ant-pagination {
  text-align: center;
  margin-top: 10px !important;
  background-color: white;
  box-shadow: 9px 10px 20px -10px #3f4a4d;
  padding-bottom: 10px !important;
}

.ant-pagination-total-text {
  display: list-item !important;
}

.result-stat-line {
  display: flex;
}
