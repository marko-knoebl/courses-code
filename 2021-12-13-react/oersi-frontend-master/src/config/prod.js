export default {
  name: "production",
  resultList: {
    component: "results",
    dataField: "name",
    pagination: true,
    showResultStats: true,
    paginationAt: "bottom",
    sortBy: null,
    pagesShow: 5,
    sizeShow: 5,
    URLParams: true,
    showEndPage: true,
    and: [
      "author",
      "license",
      "search",
      "provider",
      "learningResourceType",
      "language",
      "about",
      "sourceOrganization",
    ],
    sortByDynamic: null,
  },
  searchComponent: {
    component: "search",
    dataField: ["name", "creator.name", "description", "keywords"],
    fieldWeights: [1, 3],
    queryFormat: "and",
    fuzziness: 0,
    debounce: 100,
    autosuggest: true,
    highlight: true,
    highlightField: "keywords",
    iconPosition: "right",
    showFilter: true,
    URLParams: true,
    and: [
      "author",
      "license",
      "provider",
      "results",
      "learningResourceType",
      "language",
      "about",
      "sourceOrganization",
    ],
  },
  multiList: [
    {
      component: "about",
      dataField: "about.id",
      title: "about",
      placeholder: "about",
      filterLabel: "about",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      size: 100,
      className: "about-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "language",
        "learningResourceType",
        "sourceOrganization",
      ],
    },
    {
      component: "learningResourceType",
      dataField: "learningResourceType.id",
      title: "resourceType",
      placeholder: "resourceType",
      filterLabel: "resourceType",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "lrt-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "language",
        "sourceOrganization",
        "about",
      ],
    },
    {
      component: "license",
      dataField: "license.id",
      title: "license",
      placeholder: "license",
      filterLabel: "license",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "license-card",
      fontAwesome: "",
      URLParams: true,
      customQuery: (value, props) => {
        return value instanceof Array
          ? {
              query: {
                bool: {
                  should: [
                    ...value.map((v) => ({
                      prefix: {
                        "license.id": v,
                      },
                    })),
                    ...value.map((v) => ({
                      prefix: {
                        "license.id": v.replace("https:/", "http:/"),
                      },
                    })),
                  ],
                },
              },
            }
          : {}
      },
      defaultQuery: getPrefixAggregationQuery("license.id", [
        "https://creativecommons.org/licenses/by/",
        "https://creativecommons.org/licenses/by-sa/",
        "https://creativecommons.org/licenses/by-nd/",
        "https://creativecommons.org/licenses/by-nc-sa/",
        "https://creativecommons.org/licenses/by-nc/",
        "https://creativecommons.org/licenses/by-nc-nd/",
        "https://creativecommons.org/publicdomain/zero/",
        "https://creativecommons.org/publicdomain/mark",
      ]),
      and: [
        "author",
        "search",
        "provider",
        "results",
        "learningResourceType",
        "language",
        "about",
        "sourceOrganization",
      ],
    },
    {
      component: "author",
      dataField: "persons.name.keyword",
      nestedField: "",
      title: "author",
      placeholder: "author",
      filterLabel: "author",
      queryFormat: "and",
      showMissing: true,
      showSearch: true,
      showFilter: true,
      size: 1000,
      className: "author-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "license",
        "search",
        "provider",
        "results",
        "learningResourceType",
        "language",
        "about",
        "sourceOrganization",
      ],
    },
    {
      component: "sourceOrganization",
      dataField: "institutions.name",
      title: "organization",
      placeholder: "organization",
      filterLabel: "organization",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: true,
      size: 100,
      className: "source-type-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "language",
        "learningResourceType",
        "about",
      ],
    },
    {
      component: "language",
      dataField: "inLanguage",
      title: "language",
      placeholder: "language",
      filterLabel: "language",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "language-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "provider",
        "results",
        "learningResourceType",
        "about",
        "sourceOrganization",
      ],
    },
    {
      component: "provider",
      dataField: "mainEntityOfPage.provider.name",
      title: "provider",
      placeholder: "provider",
      filterLabel: "provider",
      queryFormat: "and",
      showMissing: true,
      showFilter: true,
      showSearch: false,
      className: "provider-card",
      fontAwesome: "",
      URLParams: true,
      and: [
        "author",
        "license",
        "search",
        "results",
        "learningResourceType",
        "language",
        "about",
        "sourceOrganization",
      ],
    },
  ],
}
function getPrefixAggregationQuery(fieldName, prefixList) {
  var aggsScript = "if (doc['" + fieldName + "'].size()==0) { return null }"
  aggsScript += prefixList.reduce(
    (result, prefix) =>
      result +
      " else if (doc['" +
      fieldName +
      "'].value.startsWith('" +
      prefix +
      "') || doc['" +
      fieldName +
      "'].value.startsWith('" +
      prefix.replace("https:/", "http:/") +
      "')) { return '" +
      prefix +
      "'}",
    ""
  )
  aggsScript += " else { return doc['" + fieldName + "'] }"
  return () => ({
    aggs: {
      "license.id": {
        terms: {
          script: {
            source: aggsScript,
            lang: "painless",
          },
        },
      },
    },
  })
}
