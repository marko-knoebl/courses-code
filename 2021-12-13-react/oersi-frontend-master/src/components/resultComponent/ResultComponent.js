import React, {useState} from "react"
import config from "react-global-configuration"
import {ReactiveList} from "@appbaseio/reactivesearch"
import "./ResultComponent.css"
import TileCard from "./card/Card"
import {withTranslation} from "react-i18next"
import "antd/lib/pagination/style/css"
import {Pagination} from "antd"
import {ConfigurationRunTime} from "../../helpers/use-context"
import getParams, {setParams} from "../../helpers/helpers"
import Grid from "@material-ui/core/Grid"

/**
 * Result Component
 * creates a Result box UI component that is used to show the result in base of search,
 * configuration fields are set in src/config/prod.json#resultList
 *
 * @author Edmond Kacaj <edmondikacaj@gmail.com>
 * @props Properties from Parent Component
 */
const ResultComponent = (props) => {
  const {setLoading, totalResult, setTotalResult} = props
  const context = React.useContext(ConfigurationRunTime)
  //declare varibale to get data from Configuration fle prod.json
  const [conf] = useState(config.get("resultList"))
  const [pageSize, setPageSize] = useState(getPageSize())
  const defaultQuery = function () {
    return {
      track_total_hits: context.TRACK_TOTAL_HITS ? context.TRACK_TOTAL_HITS : true,
    }
  }
  return (
    <>
      <ReactiveList
        componentId={conf.component}
        dataField={conf.dataField}
        stream={false}
        pagination={conf.pagination}
        paginationAt={conf.paginationAt}
        pages={conf.pagesShow}
        sortBy={conf.sortBy}
        size={pageSize}
        showLoader={false}
        showEndPage={conf.showEndPage}
        URLParams={conf.URLParams}
        showResultStats={conf.showResultStats}
        renderError
        react={{and: conf.and}}
        defaultQuery={defaultQuery}
        sortOptions={conf.sortByDynamic}
        renderNoResults={() => setTotalResult(0)}
        renderResultStats={(stats) => renderStatistics(stats)}
        renderPagination={({
          pages,
          totalPages,
          currentPage,
          setPage,
          fragmentName,
        }) => {
          return (
            <Pagination
              showQuickJumper
              current={currentPage + 1}
              defaultCurrent={currentPage + 1}
              total={totalResult}
              pageSizeOptions={context.RESULT_PAGE_SIZE_OPTIONS}
              showTotal={(total, range) =>
                props.t("RESULT_LIST.SHOW_TOTAL", {
                  rangeStart: range[0],
                  rangeEnd: range[1],
                  total: total,
                })
              }
              defaultPageSize={pageSize}
              onChange={(page, pageSiz) => {
                setPage(page - 1)
              }}
              onShowSizeChange={(current, size) => {
                setPageSize(size)
                window.location.search = setParams(window.location, {
                  name: "size",
                  value: size,
                })
              }}
            />
          )
        }}
      >
        {({data, error, loading}) => (
          <Grid container direction="row" alignItems="flex-start">
            {setLoading(loading)}
            {data.map((item) => (
              <Grid item xs={12} sm={6} md={4} lg={3} xl={2}>
                <TileCard key={item._id} {...item} />
              </Grid>
            ))}
          </Grid>
        )}
      </ReactiveList>
    </>
  )
  function renderStatistics(stats) {
    setTotalResult(stats.numberOfResults)
  }
  function getPageSize() {
    const sizeParam = getParams(window.location, "size")
    if (
      sizeParam != null &&
      context.RESULT_PAGE_SIZE_OPTIONS.indexOf(sizeParam) !== -1
    ) {
      return parseInt(sizeParam)
    } else {
      return context.NR_OF_RESULT_PER_PAGE
    }
  }
}

ResultComponent.propTypes = {}

export default withTranslation()(ResultComponent)
