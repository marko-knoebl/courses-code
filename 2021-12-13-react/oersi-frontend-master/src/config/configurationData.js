import config from "react-global-configuration"
import prod from "./prod"

export const registerConfiguration = () => {
  config.set(prod, {freeze: false})
}
