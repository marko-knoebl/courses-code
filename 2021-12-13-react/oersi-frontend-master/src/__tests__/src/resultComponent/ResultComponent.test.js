import React from "react"
import {shallow} from "../../../setupTests"
import {registerConfiguration} from "../../../config/configurationData"
import ResultComponent from "../../../components/resultComponent/ResultComponent"

describe("ResultComponent ==> Test UI  ", () => {
  registerConfiguration()
  const wrapperShadow = shallow(<ResultComponent />)
  it("ResultComponent : should render correctly", () => {
    expect(wrapperShadow).toMatchSnapshot()
  })
})
