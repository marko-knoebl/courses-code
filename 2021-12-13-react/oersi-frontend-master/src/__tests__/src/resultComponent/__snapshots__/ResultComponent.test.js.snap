// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultComponent ==> Test UI   ResultComponent : should render correctly 1`] = `ShallowWrapper {}`;
